### ISSUE TITLE
The problem with generation after training.

### ISSUE THREAD

**Issue by Ruziy** [microsoft/mattergen#93]
Good afternoon, developers. I ran into a problem. Let me tell you from the beginning. My goal was to train the model on mp_20 for fine-tuning on 2 parameters "elements/dft_poisson_ratio".  
In the first stage: I trained the unconditional model on val_loss = 0.45->99 epochs. 
At the second stage: to fine-tune for dft_poisson_ratio, I formed yaml for the float type of data.
**_target_: mattergen.property_embeddings.PropertyEmbedding
name: dft_poisson_ratio
unconditional_embedding_module:
  _target_: mattergen.property_embeddings.EmbeddingVector
  hidden_dim: ${lightning_module.diffusion_module.model.hidden_dim}
conditional_embedding_module:
  _target_: mattergen.diffusion.model_utils.NoiseLevelEncoding
  d_model: ${lightning_module.diffusion_module.model.hidden_dim}
scaler:
  _target_: mattergen.common.utils.data_utils.StandardScalerTorch**
For elements, I converted the data from ['C','Br','Fe'] to type "C-Br-Fe" to chemical_system.yaml could work with this data. An example of my elements.yaml.
**_target_: mattergen.property_embeddings.PropertyEmbedding
name: elements
unconditional_embedding_module:
  _target_: mattergen.property_embeddings.EmbeddingVector
  hidden_dim: ${lightning_module.diffusion_module.model.hidden_dim}
conditional_embedding_module:
  _target_: mattergen.property_embeddings.ChemicalSystemMultiHotEmbedding
  hidden_dim: ${lightning_module.diffusion_module.model.hidden_dim}
scaler:
  _target_: torch.nn.Identity**
After that, I started additional training on these 2 parameters (fine tuning) and got val_loss = 0.31 -> 79epoch.
At the third stage: I started testing my model for generating material on request: 
**mattergen-generate $RESULTS_PATH --model_path=$MODEL_PATH --batch_size=16 --properties_to_condition_on="{'dft_poisson_ratio': 0.290199, 'elements': 'Ni-Cr-Fe'}" --diffusion_guidance_factor=2.0**
Analyzing the results, I came to a strong disappointment, when I looked at OVITO generated_crystals_cif, I saw that on the very first cif. The model does not contain the necessary Ni-Cr-Fe, but a completely random lattice.

![Image](https://github.com/user-attachments/assets/b0945eb8-dc68-4114-bda8-f45478219e59)

To test the feasibility of my task, I decided to ask the same query for your chemical_system model. And everything was fine.

![Image](https://github.com/user-attachments/assets/27b9a8c2-a82e-4c42-af57-86e7d0932c8a)

Could you tell me a possible problem or a mistake that I made at the time of learning?

**Comment by Ruziy** [comment]
upd. Perhaps the course of my reasoning will give you more understanding. Initially, I thought that the problem of the relationship between the two parameters was that it was difficult to find a material that corresponded to my chemical composition, in this case elements, and was close to the set value of the Poisson's ratio. But when generating an element without specifying the Poisson's ratio, I also got a random chemical composition at the output. 
After that, I thought that the problem was in the data itself in the "elements" table. That the dataset has the form "elements".type->list, and the fine-tuning has "elements".type->str. What is stated in the <code>. Therefore, I quickly trained the model immediately on a dataset of the "elements" format.type->str in 50 epoch I achieved loss_value = 0.49 for the unconditional model. And I achieved an accuracy of los_val = 0.35 for fine tuning in 25 epochs. But the problem remains (the first image). 
The only thing that comes to mind now is that the los_val metric does not demonstrate the accuracy of the model, and it is necessary to train it to ~200/250epoch to be sure of its training / or something is wrong with yaml files, but this is too long-term a process and therefore I would like expert opinions on this matter.

**Comment by Ruziy** [comment]
upd.2 In the course of solving this problem, a question arose since alex_mp_20 actually contains mp_20, and in turn mp_20 contains information about the Poisson ratio for the material. Is there an alex_mp_20 that contains this data? You've probably already transformed the dataset for your tasks and it doesn't have its original appearance.
upd.3 I also came across the hypothesis that I am using data_module incorrectly. It is possible for fine-tuning in data_module.yaml in properties: [] is it necessary to specify parameters for fine-tuning?

**Comment by txie-93** [comment]
Have you followed the instructions in our README [here](https://github.com/microsoft/mattergen?tab=readme-ov-file#multi-property-fine-tuning) to fine-tune the model jointly on chemical system and dft_poisson_ratio? It seems that you only fine-tuned it on single property (dft_poisson_ratio). It might explain why you cannot generate conditioning on chemical system. 

Also, I'd recommend you fine-tune directly from our provided [base model](https://github.com/microsoft/mattergen/tree/main/checkpoints/mattergen_base). Training the base model on MP20 for 99 epochs is likely not enough.

**Comment by txie-93** [comment]
After skimming your previous conversation with @danielzuegner in #82, I make the same recommendation to replace "elements" with "chemical_system", and redo the multi-property fine-tuning. This way, you can reuse our code to fine-tune given chemical system. Likely you have some bug with your own implementation to condition on "elements".

**Comment by Ruziy** [comment]
@txie-93 OK, thanks, I'll try it. Due to your prompt responses, could you please answer 1 more. I am interested in a question regarding the generation results. Or rather, the files that we receive at the exit.  I am interested in generated_crystals_cif why 0-15 cif files are used there, what progress do they characterize? Does the generated_crystals.extxyz file represent the final generated material?   I am also interested in information about the contents of *.cif files. Like this:
**data_image0
_chemical_formula_structural       FeCrFe2Cr2FeNiFe3NiFe2NiFe3Ni2
_chemical_formula_sum              "Fe12 Cr3 Ni5"
_cell_length_a       3.074340190589395
_cell_length_b       7.680496805312162
_cell_length_c       9.790635108947754
_cell_angle_alpha    101.09854042057356
_cell_angle_beta     89.95864571115504
_cell_angle_gamma    89.98781744561214
_space_group_name_H-M_alt    "P 1"
_space_group_IT_number       1
loop_
  _space_group_symop_operation_xyz
  'x, y, z'
loop_
  _atom_site_type_symbol
  _atom_site_label
  _atom_site_symmetry_multiplicity
  _atom_site_fract_x
  _atom_site_fract_y
  _atom_site_fract_z
  _atom_site_occupancy
Fe  Fe1       1.0  0.9581835269927979  0.1665475070476532  0.7390089631080627  1.0000**
Also for the content .extxyz for example:
**4
Lattice="3.5554122924804688 0.0 0.0015426157042384148 -0.001068046811986971 3.55414279148726 -0.0005145111936144531 0.0 0.0 3.553982973098755" Properties=species:S:1:pos:R:3 pbc="T T T"
Cr       0.71375895       2.65317909       0.60386655
Fe       0.71389246       0.87694500       2.38208662
Ni       2.49157741       0.87589426       0.60530258
Fe       2.49108439       2.65402710       2.38197514**
Thanks.

**Comment by Ruziy** [comment]
@txie-93 @danielzuegner I also ask you to answer these questions. upd.2 In the course of solving this problem, a question arose since alex_mp_20 actually contains mp_20, and in turn mp_20 contains information about the Poisson ratio for the material. Is there an alex_mp_20 that contains this data? You've probably already transformed the dataset for your tasks and it doesn't have its original appearance.
upd.3 I also came across the hypothesis that I am using data_module incorrectly. It is possible for fine-tuning in data_module.yaml in properties: [] is it necessary to specify parameters for fine-tuning?

**Comment by danielzuegner** [comment]
@Ruziy you can copy the Poisson ratio information from mp20 to alex_mp_20. We don't have a fully labeled version of alex_mp_20 with Poisson ratio. The identifier information in the datasets should be sufficient to copy the information over.

In the fine-tuning command from the README, we do provide the config override `data_module.properties=["$PROPERTY"]`, which adds the respective property to the data module.

**Comment by Ruziy** [comment]
@danielzuegner thanks, i think you can close this issue.
